web/src/components/FriendGroup/FriendGroup.jsx

```diff
@@ -0,0 +1,146 @@
+import { useAuth0 } from "@auth0/auth0-react";
+import { useState, useEffect } from "react";
+import * as React from "react";
+import Api from "../../helpers/api";
+import Button from "../Common/Button";
+import Popup from "../Common/Popup";
+
+const FriendGroup = () => {
+  const { user, getAccessTokenSilently } = useAuth0();
+  const [group_name, setGroupName] = useState("");
+  const [userFriendGroups, setUserFriendGroups] = useState("");
+  const [userFriendGroupsSize, setUserFriendGroupsSize] = useState("");
+  const [showModal, setShowModal] = useState(false);
+  const [showPopupModal, setShowPopupModal] = useState(false);
+  const [modalPopupTitle, setModalPopupTitle] = useState(false);
+  const [modalPopupDescription, setModalPopupDescription] = useState(false);
+
+  const api = new Api();
+
+  useEffect(() => {
+    try {
+      getFriendGroups();
+    } catch (err) {
+      console.log("error" + err);
+    }
+  }, []);
+
+
+  async function getFriendGroups() {
+    try {
+      api.getFriendGroups(
+        await getAccessTokenSilently()
+      )
+        .then((res) => {
+          // console.log("result from get friendGroups message: " + res.message)
+          // console.log("result from get friendGroups profile: " + res.profile[0].group_name)
+          // setUserFriendGroupsSize(res.profile.length)
+          setUserFriendGroups(res.profile);
+        });
+    } catch (err) {
+      console.log("error" + err);
+    }
+    // add error handling for friendNotFound
+    handleModalOpen();
+  }
+
+  //creates friendGroup
+  async function createFriendGroup(event) {
+    event.preventDefault(); //prevent page refresh
+    console.log("The group name being created is: " + group_name);
+    await api
+      .createFriendGroup(
+        await getAccessTokenSilently(),
+        group_name
+      )
+      .then((res) => {
+        console.log(res);
+        if (res.data.message == 'Successfully posted friend group') { //it's a valid group
+          console.log("data content: " + res.data);
+        } else {
+          console.log("Nothing created, there must be a mistake!");
+        }
+      });
+    setShowModal(false);
+    getFriendGroups(); // get all groups so we can update the display
+  }
+
+
+  const handleModalOpen = () => {
+    setShowModal(true);
+  };
+
+  return (
+    <div className="friend-group">
+      {userFriendGroups ? (
+        <div>
+          <h1 className="text-white"> My {userFriendGroupsSize} friend groups: </h1>
```

Comment:
The group size does not get displayed here because I don't think you set it in the code, but this really does not matter. Just letting you know in case you really wanted to have the size displayed, otherwise we can just ignore it

Reply:
It’s set with setUserFriendGroups inside the getFriendGroups method ! I think it should be good it was working locally